src/wallet/wallet.h

```diff
@@ -141,9 +142,18 @@ class CKeyPool
                    (this will be the case for any wallet before the HD chain split version) */
                 fInternal = false;
             }
+            try {
+                READWRITE(m_pre_split);
```

Comment:
You can test for this using `!(ser_action.ForRead() && s.empty())` instead of trying/catching.

Reply:
I don't quite understand what you mean.

Also, I would like to keep this consistent with the block above which does the same thing.

Reply:
I think sipa meant to check during read if there are more bytes in the stream because this changed only appends data at the end.

Reply:
Indeed, what @jonasschnelli says. If it's how similar already operates we can change it later in one go, too.